File: src/main/kotlin/com/example/testDouble/mockk/Store.kt
package com.example.testDouble.mockk

import org.springframework.stereotype.Component

interface Store {
    fun searchItem(itemName: String): String
}

@Component
class StoreImpl(
    private val backyard: Backyard
) : Store {
    override fun searchItem(itemName: String): String {
        return backyard.findItem(itemName)
    }
}

File: react/src/main/Store.ts
import BackyardImpl, { Backyard } from '@/main/Backyard.ts'

interface Store {
  searchItem(itemName: string): string
}


export default class StoreImpl implements Store {
  backyard: Backyard

  constructor(backyard: Backyard = new BackyardImpl()) {
    this.backyard = backyard
  }

  searchItem(itemName: string): string {
    return this.backyard.findItem(itemName)
  }
}
